extensions/markdown-language-features/src/languageFeatures/references.ts

```diff
@@ -341,8 +341,8 @@ export async function tryResolveLinkPath(originalUri: vscode.Uri, workspace: IMd
 		return originalUri;
 	}
 
-	// We don't think the file exists. If it doesn't already have an extension, try tacking on a `.md` and using that instead
-	if (uri.Utils.extname(originalUri) === '') {
+	// We don't think the file exists. If it doesn't already have a `.md` extension, try tacking on a `.md` and using that instead
+	if (uri.Utils.extname(originalUri) !== '.md') {
```

Comment:
This seems ok. You can use the `looksLikeMarkdownPath` helper for this to handle other md file extensions too: https://github.com/microsoft/vscode/blob/0bc31097613ab4e16c1e4dda85750bc0326e77e6/extensions/markdown-language-features/src/util/file.ts#L25

Maybe we should also skip this check for common image file extensions (`.jpg`, `.png`, `.gif`, ...)

Reply:
Thanks for the hint. I've added a `lookLikeImagePath` helper function to exclude image files too.